astropy/io/fits/file.py

```diff
@@ -436,10 +440,14 @@ def _open_filename(self, filename, mode, clobber):
         elif ext == '.zip' or magic.startswith(PKZIP_MAGIC):
             # Handle zip files
             self._open_zipfile(self.name, mode)
+        elif ext == '.bz2' or magic.startswith(BZIP2_MAGIC):
+            # Handle bzip2 files
+            self._file = bz2.BZ2File(self.name, PYFITS_MODES[mode])
+            self.compression = 'bzip2'
         else:
             self._file = fileobj_open(self.name, PYFITS_MODES[mode])
             # Make certain we're back at the beginning of the file
-        self._file.seek(0)
+            self._file.seek(0)
```

Comment:
Ah, I see what you're saying about the indentation here.  This line should be at the level of indentation it was at previously--we should try seeking to the beginning in all cases (compressed and uncompressed).  It's the comment that was mis-indented. 

It would probably be clearer if there were a newline before the comment, and the comment were dedented one level.